index.js

```diff
@@ -104,6 +140,49 @@ class Client {
     }
     return analyses
   }
+
+  /**
+    * Runs MythX analysis return issue information and metadata regarding the run
+    *
+    * @param {options} object - structure which must contain:
+    *      {data} object       - information containing Smart Contract information to be analyzed
+    *      {timeout} number    - optional timeout value in milliseconds
+    *
+    * @returns object which contains:
+    *      (issues}  object - an like object which of issues is grouped by (file) input container.
+    *      {status}  object - status information as returned in each object of analyses().
+    *
+    **/
+  async analyzeWithStatus (options) {
+    const issues = await this.analyze(options)
+    const uuid = issues.uuid
+    delete issues.uuid
+    const status = await this.getStatus(uuid)
+    return {
+      issues,
+      status
+    }
+  }
+
+  async getStatus (uuid, inputApiUrl = defaultApiUrl) {
+    let accessToken = this.accessToken
+    if (!accessToken) {
+      const tokens = await login.do(this.email, this.ethAddress, this.password, this.apiUrl)
+      accessToken = tokens.access
+    }
+    const url = `${inputApiUrl}/${defaultApiVersion}/analyses/${uuid}`
+    return simpleRequester.do({ url, accessToken: accessToken, json: true })
+  }
+
+  async getIssues (uuid, inputApiUrl = defaultApiUrl) {
+    let accessToken = this.accessToken
```

Comment:
Why not using `this.accessToken` here? With the local variable, if the same client does an additional request it will need to login again; if we save the token in `this.accessToken`, subsequent requests won't need to login again.

Reply:
Good idea. I belive this is covered in 9e151a1